src/Path.hs

```diff
@@ -379,6 +383,12 @@ dirname (Path l) =
 --------------------------------------------------------------------------------
 -- Internal functions
 
+-- | On Posix root is always "/" but on windows it could be "C:" or "C:\" etc.
+-- So check in a portable manner.
+isRootDir :: FilePath -> Bool
+isRootDir p = p' == FilePath.takeDirectory p'
+    where p' = FilePath.dropTrailingPathSeparator p
+
```

Comment:
Perhaps http://hackage.haskell.org/package/filepath-1.4.1.0/docs/System-FilePath-Posix.html#v:isDrive can achieve this?